src/clj_puppetdb/vcr.clj

```diff
@@ -0,0 +1,28 @@
+(ns clj-puppetdb.vcr
+  (:require [clojure.edn :as edn]
+            [me.raynes.fs :as fs]
+            [puppetlabs.kitchensink.core :refer [utf8-string->sha1]]
+            [puppetlabs.http.client.sync :as http]
+            [puppetlabs.kitchensink.core :refer [dissoc-in]]))
+
+(defn- vcr-filename
+  [vcr-dir path]
+  (str vcr-dir "/" (utf8-string->sha1 path) ".clj"))
```

Comment:
In this "case" path is the actual HTTP get path, not the PDB query endpoint. It already has the query params attached so different queries result in different hashes. The query params are also sorted so irrespective of the way they are supplied the path should be stable.

I updated the tests to verify both considerations though.